src/beta_inc.jl

```diff
@@ -0,0 +1,945 @@
+
+function apser(a::Float64, b::Float64, x::Float64, epps::Float64)         
+    g = Base.MathConstants.eulergamma
+    bx = b*x
+    t = x - bx
+    if b*epps > 2e-2
+        c = log(bx) + g + t
+    else
+        c = log(x) + digamma(b) + g + t
+    end
+
+
+    tol = 5.0*epps*abs(c)
+    j = 1.0
+    s = 0.0
+
+    j += 1.0
+    t *= (x - bx/j)
+    aj = t/j
+    s += aj
+    while abs(aj) > tol
+       j += 1.0
+       t *= (x - bx/j)
+       aj = t/j
+       s += aj
+    end
+    return -a*(c + s)
+end
+
+#B .LE. 1 OR B*X .LE. 0.7
+
+function bpser(a::Float64, b::Float64, x::Float64, epps::Float64)
+    ans = 0.0
+    if x == 0.0
+        return 0.0
+    end
+    a0 = min(a,b)
+    if a0 < 1.0
+      @goto l10
+    end
+    z = a*log(x) - logbeta(a,b)
+    ans = exp(z)/a
+
+    @label l10
+     b0 = max(a,b)
+     if b0 >= 8.0
+        @goto l60
+     elseif b0 > 1.0
+        @goto l40
+     end
+     #PROCEDURE FOR A0 < 1 && B0 < 1
+     ans = x^a
+     if ans == 0.0
+        return ans
+     end
+     apb = a + b
+     if apb > 1.0
+        @goto l20
+     else
+        z = 1.0 + SpecialFunctions.rgamma1pm1(apb)
+        @goto l30
+     end
+    @label l20
+     u = a + b - 1.0
+     z = (1.0 + SpecialFunctions.rgamma1pm1(u))/apb
+    @label l30
+     c = (1.0 + SpecialFunctions.rgamma1pm1(a))*(1.0 + SpecialFunctions.rgamma1pm1(b))/z
+     ans *= c*(b/apb)
+     @goto l70
+
+    #PROCEDURE FOR A0 < 1 AND 1 < B0 <8
+    @label l40
+     u = SpecialFunctions.loggamma1p(a0)
+     m = b0 - 1.0
+     if m < 1.0
+        @goto l50
+     end
+     c = 1.0
+    @label l41
+     for i = 1:m
+        b0 -= 1.0
+        c *= (b0/(a0+b0))
+     end
+     u += log(c)
+
+    @label l50
+     z = a*log(x) - u
+     b0 -= 1.0
+     apb = a0 + b0
+     if apb > 1.0
+        @goto l51
+     else
+        t = 1.0 + SpecialFunctions.rgamma1pm1(apb)
+        @goto l52
+     end
+    @label l51
+     u = a0 + b0 - 1.0
+     t = (1.0 + SpecialFunctions.rgamma1pm1(u))/apb
+    @label l52
+     ans = exp(z)*(a0/a)*(1.0 + SpecialFunctions.rgamma1pm1(b0))/t
+     @goto l70
+
+    #PROCEDURE FOR A0 < 1 AND B0 >= 8
+
+    @label l60
+     u = SpecialFunctions.rgamma1pm1(a0) + loggammadiv(a0,b0)
+     z = a*log(x) - u
+     ans = (a0/a)*exp(z)
+    @label l70
+     if ans == 0.0 || a <= 0.1*epps
+        return ans
+     end
+
+    # COMPUTE THE SERIES
+
+    sm = 0.0
+    n = 0.0
+    c = 1.0
+    tol = epps/a
+    @label l100
+     n += 1.0
+     c *= x*(1.0 - b/n)
+     w = c/(a + n)
+     sm += w
+     while abs(w) > tol
+         n += 1.0
+         c *= x*(1.0 - b/n)
+         w = c/(a+n)
+         sm += w
+     end
+    return ans*(1.0 + a*sm)
+end
+
+"""
+    bup(a,b,x,y,n,epps)
+
+Compute ``I_{x}(a,b) - I_{x}(a+n,b)`` where n is positive integer and epps is tolerance.
+"""
+function bup(a::Float64, b::Float64, x::Float64, y::Float64, n::Integer, epps::Float64)
+    apb = a + b
+    ap1 = a + 1.0
+    mu = 0.0
+    d = 1.0
+    if n == 1 || a < 1.0
```

Comment:
rewrite to avoid `@goto l10`:
```
if n != 1 && a >= 1.0
        mu = abs(exparg_n)
        k = exparg_p
        if k < mu 
            mu = k
        end
        t = mu
        d = exp(-t)
end
```